File: 2020_2021/L3_S5_2020-2021/INFO0503_modélisation web serv & web avancé/INFO503/TP2_sofiane/Place.php
<?php
class Place{

    // attributs

    private $rangee;
    private $numPlace;
    private $estVide;
    private $voiture;

    // constructeur par initialisation

    public function __construct($rangee,$numPlace){
    	$this->rangee = $rangee;
        $this->numPlace = $numPlace;
        $this->estVide = false;
        $this->voiture = null;
    }
    // getters

    public function getRangee(){
        return $this->rangee;
    }

    public function getNumPlace(){
        return $this->numPlace;
    }

    public function getEstVide(){
        return $this->estVide;
    }

    public function getVoiture(){
        return $this->voiture;
    }

    public function setVoiture($voiture){
        $this->voiture = $voiture;
        $this->estVide = true;
    }

    public function enleverVoiture(){
        if(!$this->estVide){
            return null;
        }
        $voiture = $this->$this->getVoiture();
        $this->voiture = null;
        $this->estVide = false;
        return null;
    }

    // methode toString

    public function __toString(){
        return "Place [rangee=" . $this->rangee . ", num=" . $this->numPlace . ", estVide=" . $this->estVide . "]";  //. ($this->estVide?($this->estVide . " ; Voiture : " . $this->voiture->__toString() . "]"):($this->estVide + "]"));
    }
}

File: 2020_2021/L3_S5_2020-2021/INFO0503_modélisation web serv & web avancé/INFO503/TP5_Projet_1_01-11-2020/Projet1/src/java/Place.java
public class Place{
    private char rangee;
    private int numPlace;
    private boolean estVide;
    private Voiture voiture;

    public Place(char rangee, int numPlace) {
        this.rangee = rangee;
        this.numPlace = numPlace;
        this.estVide = false;
        this.voiture = null;
    }
    public Place(char rangee, int num, Voiture voiture) {
        this.rangee = rangee;
        this.numPlace = numPlace;
        this.voiture = voiture;
        this.estVide = true;
    }
    public Place(Place pl) {
        this.rangee = pl.rangee;
        this.numPlace = pl.numPlace;
        this.estVide = pl.estVide;
        this.voiture = pl.voiture;
    }

    public char getRangee() {return rangee;}
    public int getNumPlace() {return numPlace;}
    public boolean estvide() {return estVide;}
    public Voiture getVoiture() {return voiture;}

    public void setVoiture(Voiture voiture) {
        this.voiture = voiture;
        this.estVide = true;
    }

    public Voiture enleverVoiture() {
        if(!this.estVide) return null;

        Voiture voiture = this.getVoiture();
        this.voiture = null;
        this.estVide = false;
        return voiture;
    }

    @Override
    public String toString() {
        return "Place [rangee=" + rangee + ", num=" + numPlace + ", estVide=" + (this.estVide?(this.estVide+ " ; Voiture : " + this.voiture.toString() + "]"):(this.estVide + "]"));
    }
}